iwyu.cc

```diff
@@ -1475,6 +1475,37 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
     return true;
   }
 
+  // Returns the first type that is not a typedef in a template.  For example,
+  // for template
+  //
+  //   template<typename T> class Foo {
+  //     typedef T value_type;
+  //     typedef value_type& reference;
+  //   };
+  //
+  // for type 'reference' it will return type T with which Foo was instantiated.
+  const Type* GetFirstNonTemplateTypedef(const TypedefType* typedef_type) {
```

Comment:
The name is a little clunky. Isn't this a form of desugaring? `DesugarTemplateTypedef` sounds more decisive :-)